### PR TITLE
ddl_notifier: delete event after all handlers finish processing

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #55722

Problem Summary:

### What changed and how does it work?

as title

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by fzzf678 — APPROVED** [review]

File: pkg/ddl/notifier/subscribe.go [hunk 1]
```diff
@@ -189,10 +212,18 @@ func (n *ddlNotifier) processEventForHandler(
 		}
 	}()
 
-	// TODO: Should we attach a timeout to this ctx?
+	now := time.Now()
 	if err = handler(ctx, n.ownedSCtx, change.event); err != nil {
 		return errors.Trace(err)
 	}
+	if time.Since(now) > logSlowProcess {
+		logutil.Logger(ctx).Warn("Slow process event",
```
Comment (time-and-fate) [inline pkg/ddl/notifier/subscribe.go:220]: I think it's better to have a label like `[ddl_notifier]` for all logs in this module.

File: pkg/ddl/notifier/subscribe.go [hunk 2]
```diff
@@ -160,12 +168,27 @@ func (n *ddlNotifier) processEvents(ctx context.Context) error {
...
+					zap.Int64("multiSchemaChangeSeq", change.multiSchemaChangeSeq),
+					zap.Error(err2))
+			}
+		}
 	}
 
 	return nil
 }
 
+const logSlowProcess = time.Second * 5
```
Comment (time-and-fate) [inline pkg/ddl/notifier/subscribe.go:190]: I think names like `slowHandlerLogThreshold` might be better.

File: pkg/ddl/notifier/store.go [hunk 3]
```diff
@@ -82,15 +82,26 @@ func (t *tableStore) UpdateProcessed(
...
+	}
+	sql := fmt.Sprintf(`
+		DELETE FROM %s.%s
+		WHERE ddl_job_id = %d AND multi_schema_change_seq = %d`,
+		t.db, t.table,
+		ddlJobID, multiSchemaChangeID)
+	if _, err := se.Execute(ctx, sql, "ddl_notifier"); err != nil {
+		return errors.Trace(err)
+	}
+	return errors.Trace(se.Commit(ctx))
```
Comment (time-and-fate) [inline pkg/ddl/notifier/store.go:102]: Should we call `Rollback()` when it fails like we did in `processEventForHandler()`?

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**